markdown/util.py

```diff
@@ -45,60 +44,269 @@
 """
 
 
-class _BlockLevelElements(set):
-    # ----------------------------------
-    # Methods common to `list` and `set`
-    # ----------------------------------
-    def copy(self) -> set[str]:
-        return _BlockLevelElements(super().copy())
+# TODO: Raise errors from list methods in the future.
+# Later, remove this class entirely and use a regular set.
+class _BlockLevelElements:
+    def __init__(self, elements: list[str]) -> None:
+        self._list = elements.copy()
+        self._set = set(self._list)
+
+    def __add__(self, other: list[str]) -> list[str]:
+        warnings.warn(
+            "Using block level elements as a list is deprecated, use it as a set instead.",
+            DeprecationWarning,
+        )
```

Comment:
I emit the same warning every time, so that users only get it once. The message goes straight to the point: don't use it as a list.